core/arch/arm/mm/core_mmu.c

```diff
@@ -1302,6 +1302,84 @@ static void set_pg_region(struct core_mmu_table_info *dir_info,
 	}
 }
 
+static bool can_map_at_level(paddr_t paddr, vaddr_t vaddr,
```

Comment:
I think this function would be easier to follow if you returned `false` as soon as it can't return `true` any longer.